### PR TITLE
feat: make Docs styling more granular

### PR DESCRIPTION
**Pull request by mmiask**
This is needed for https://github.com/stoplightio/platform-internal/issues/6253 in order to hide headings.

Aside from that, it make the styling of Docs a bit more granular and removes unnecessary logic in `HttpService`

The visual outcome is the same as before, this only allows a more precise manipulation of styling.

### CODE REVIEW

File: packages/elements-core/src/components/Docs/HttpOperation/HttpOperation.tsx [hunk 1]
```diff
@@ -35,73 +43,56 @@ const HttpOperationComponent = React.memo<HttpOperationProps>(
 
     const hasBadges = isDeprecated || securitySchemes.length > 0 || isInternal;
 
-    if (!headless)
-      return (
-        <Box bg="transparent" className={cn('HttpOperation', className)} w="full">
+    return (
+      <Box bg="transparent" className={cn('HttpOperation', className)} w="full">
+        {!customStyle.noHeading && (
```
Comment (marcelltoth) [inline packages/elements-core/src/components/Docs/HttpOperation/HttpOperation.tsx:48]: Wow this is much nicer!

File: packages/elements-core/src/components/Docs/HttpOperation/HttpOperation.tsx [hunk 2]
```diff
@@ -18,7 +18,15 @@ import { Responses } from './Responses';
...
 const HttpOperationComponent = React.memo<HttpOperationProps>(
-  ({ className, data: unresolvedData, headless, uri, hideTryIt, hideTryItPanel, allowRouting = false }) => {
+  ({
+    className,
+    data: unresolvedData,
+    uri,
+    hideTryIt,
+    hideTryItPanel,
+    allowRouting = false,
+    customStyle = { noHeading: false },
```
Comment (marcelltoth) [inline packages/elements-core/src/components/Docs/HttpOperation/HttpOperation.tsx:28]: Unfortunately you can't default options like this. The problem is that anyone who passes even one prop will override this. I mean if `customStyle={{asd: true}}` then you lose `noHeading: false`.

The solution is usually to use [_.defaults](https://lodash.com/docs/4.17.15#defaults) inside the function.

Comment (marcelltoth) [reply packages/elements-core/src/components/Docs/HttpOperation/HttpOperation.tsx:28]: Also you don't need that default as your prop is negative. Undefined acts pretty much the same, it's also falsey.

File: packages/elements-core/src/components/Docs/Docs.tsx [hunk 3]
```diff
@@ -50,6 +50,22 @@ interface BaseDocsProps {
    * Allows to use internal routing (requires wrapping with Router component)
    */
   allowRouting?: boolean;
+
+  /**
+   * Allows to customize the layout of Docs
+   */
+  customStyle?: {
```
[the remark this answers is not in the document]
Comment (mmiask) [reply packages/elements-core/src/components/Docs/Docs.tsx:57]: That's an absolutely valid point. I'm not happy with `customStyle` as well, `styleOptions` or `layoutOptions` will be better

[the remark this answers is not in the document]
Comment (marcelltoth) [reply packages/elements-core/src/components/Docs/Docs.tsx:57]: `layoutOptions` is fine by me, too. I wrote these before I read your discussion

File: packages/elements-core/src/components/Docs/HttpOperation/HttpOperation.tsx (hunk 2 above)
Comment (mmiask) [reply packages/elements-core/src/components/Docs/HttpOperation/HttpOperation.tsx]: Also, I will change `hidePoweredBy` to `showPoweredBy`- first of all hiding is the default behavior, and secondly it allows to get rid of setting these default values

**Review by marcelltoth — APPROVED** [review]
Looks good now!

**Pull request merged**